### DISCUSSION TITLE
A SUBJECT with with multiple subscribers - notify to only intended subscriber

### DISCUSSION THREAD

**Discussion by rdkumavat1** [nats-io/nats-server#5343] [General]
I have a subject with multiple subscribers, the subject may contain multiple types of messages for subscribers. I need to know the way how specific subscriber will consume messages from the common subject and other subscribers will not.  For ex - I have a subject SUBJECT1 with active subscribers - S1 and S2. SUBJECT1 has a new message for subscriber S1, what is the configuration that will dispatch a message to S1 only and not to S2 and vice versa when SUBJECT1 has a message for subscriber S2?

**Comment by ripienaar** [comment]
I think the only real way to achieve such granular delivery is to create seperate subjects by type of message, something like `events.s1` and then S1 subscribe subscribe to that specific subject.

We have no ability to route on headers or content of the message, only subject.

**Comment by zro-gaia** [comment]
You can configure queueGroups to segment subscriber pools
